src/main/main.c

```diff
@@ -70,6 +71,9 @@ enum {
 
 /** File descriptor handler struct */
 struct fhs {
+	struct le he;        /**< Hash entry                        */
+	struct le le; 	     /**< Cache/Delete entry                */
+	int index;           /**< Index used for arrays             */
```

Comment:
this will add quite a large overhead to each FD element.
sizeof struct le is 64 bytes, so times two is 128 bytes.

If the number of FDs is 100.000 the overhead would be 12.8 Mbytes

Reply:
Note that you're comparing the overhead in terms of megabytes in one case vs. the overhead in terms of gigabytes in another. See the discussion in https://github.com/baresip/re/pull/417.

Reply:
I combined the `struct le` entries now:

```c
$ pahole -Cfhs build/libre.a   
struct fhs {
        struct le                  le;                   /*     0    32 */
        re_sock_t                  fd;                   /*    32     4 */
        int                        flags;                /*    36     4 */
        fd_h *                     fh;                   /*    40     8 */
        void *                     arg;                  /*    48     8 */
        _Bool                      active;               /*    56     1 */

        /* size: 64, cachelines: 1, members: 6 */
        /* padding: 7 */
};
```